### DISCUSSION TITLE
Env expand using vercel environment variables not working

### DISCUSSION THREAD

**Discussion by iamarpitpatidar** [vercel/community#3997] [Help] [answered]
### Page to Investigate

https://vercel.com/iamarpitpatidar/PROJECT_NAME/settings/environment-variables

### Steps to Reproduce
- Add a environment variable with value `anything$NEXT_PUBLIC_VERCEL_URL`
- I'm using nextjs13 app directory so maybe create a page and print the variable
- instead of printing the vercel_deploy url with print `anything$NEXT_PUBLIC_VERCEL_URL`

### Similar discussions
- I've searched for similar discussions
- found a outdated discussion with solution I should escape the \$ but if I do it won't work coz then `nextjs` won't treat it as expanding variable

**Comment by amyegan** [comment, marked as answer]
Hey @iamarpitpatidar. [Referencing other variables](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#referencing-other-variables) is a Next.js feature specific to `.env*` files and is not supported inside Vercel [Environment Variables](https://vercel.com/docs/projects/environment-variables#). You'll need to use a `.env` file to reference other variables like that.

**Reply by iamarpitpatidar** [reply]
But then those variables won't be secret no ? @amyegan

**Reply by droplab** [reply]
Yes, this feels like a miss on the Vercel side. Using an .env file in production is a major no-no. 

Are there any plans to support environment variable expansion on the roadmap?

**Reply by ADTC** [reply]
> Using an .env file in production is a major no-no.

It's fine as long as your team knows. For example, have adequate warnings in `.env.production` file, like this:

```env
# DO NOT place any other environment variables here.
# All other environment variables should be set in Vercel.
# This variable is here so that the system variable can be resolved.

# This variable is applicable for Preview and Development only.
# For production, AUTH0_BASE_URL should be set in Vercel, which overrides this.
NEXT_PUBLIC_AUTH0_BASE_URL=$VERCEL_BRANCH_URL
```

Note that we only set the "safe" variables that need to be resolved from system variables in the file, which is committed in the repo. All secrets should still be in Vercel (not committed to repo).
